rumqttd/src/router/routing.rs

```diff
@@ -1129,15 +1242,44 @@ fn forward_device_data(
         return ConsumeStatus::FilterCaughtup;
     }
 
+    let mut topic_alias = broker_topic_aliases
+        .as_ref()
+        .and_then(|aliases| aliases.get_alias(&request.filter));
+
+    let topic_alias_exists = topic_alias.is_some();
+
+    // if topic alias doesn't exists, try creating new one!
+    if !topic_alias_exists {
+        topic_alias = broker_topic_aliases
+            .as_mut()
+            .and_then(|broker_aliases| broker_aliases.set_new_alias(&request.filter))
+    }
```

Comment:
i think `topic_alias_exists` should be renamed to `topic_alias_already_existed` or something more descriptive. because in the `if` branch we may create a topic alias and this variable name wouldn't reflect what the actual state is.